src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -260,7 +260,7 @@ public override void VisitBinaryExpression(BinaryExpressionSyntax node)
         public override void VisitDeclarationExpression(DeclarationExpressionSyntax node)
         {
             var argumentSyntax = (ArgumentSyntax)node?.Parent;
```

Comment:
>var argumentSyntax = (ArgumentSyntax)node?.Parent; [](start = 12, length = 50)

I don't think we ever expect the ```node``` to be null here and the code below doesn't expect ```argumentSyntax``` to be null, please remove the conditional access.